stylix/palette.nix

```diff
@@ -5,27 +5,129 @@
   options,
   ...
 }:
-
 let
   cfg = config.stylix;
-
+  adjustLightness =
+    rgbColorString: primaryScale:
+    let
+      stripped = builtins.replaceStrings [ "rgb(" ")" ] [ "" "" ] rgbColorString;
+      values = builtins.split "," stripped;
+      v1 = builtins.fromJSON (builtins.elemAt values 0);
+      v2 = builtins.fromJSON (builtins.elemAt values 2);
+      v3 = builtins.fromJSON (builtins.elemAt values 4);
+      preLightness = (v1 + v2 + v3) / 3.0;
+      adj =
+        (preLightness / 255.0 * (1.0 - primaryScale) + primaryScale) / preLightness
+        * 255.0;
+      v1adj = lib.max (lib.min (v1 * adj) 255.0) 0.0;
+      v2adj = lib.max (lib.min (v2 * adj) 255.0) 0.0;
+      v3adj = lib.max (lib.min (v3 * adj) 255.0) 0.0;
+      round =
+        x:
+        let
+          floored = builtins.floor x;
+          diff = x - floored;
+        in
+        if diff >= 0.5 then floored + 1 else floored;
+    in
+    builtins.toString (
+      lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v1adj))
+    )
+    + builtins.toString (
+      lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v2adj))
+    )
+    + builtins.toString (
+      lib.strings.fixedWidthString 2 "0" (lib.toHexString (round v3adj))
+    );
 in
 {
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "stylix"
+        "polarity"
+      ]
+      "The stylix.polarity option has been removed as a new palette generation backend is now used.\nNew options for this palette generation backend are available at stylix.themeGeneration.\nThe new stylix.themeGeneration.polarity sets the generated theme polarity (\"dark\", \"light\" options) and the \"either\" option doesn't exist anymore.\nYou have to explicitly choose what polarity you want.\nThe new options, stylix.themeGeneration.scheme, stylix.themeGeneration.contrast, stylix.themeGeneration.primaryScale.light, and stylix.themeGeneration.primaryScale.dark, let you tweak the generated color scheme further if need be."
```

Comment:
It might be better to put the message on a single line, since it improves `grep`ability and groups errors on a line basis.